### DISCUSSION TITLE
MediaElement: To add metadata support to .NET MAUI

### DISCUSSION THREAD

**Discussion by imadofficial** [CommunityToolkit/Maui#1682] [New Feature Discussions]
## Introduction
Having MediaElement is already a huge step towards having a matured framework. It already works really well thanks to Gerald & the many different contributions people have made over the last few months.

## Future proposal
Having metadata show up when playing anything is a crucial part of having an app that works really well. It gives people the ability to see what they're playing at a glans whether it being Windows, macOS or even Android. If you guys are wondering here are some reference images down below of what i'm talking about.

<p float="left">
  <img src="https://github.com/CommunityToolkit/Maui/assets/59332263/1e3ce6a2-4104-4dbc-839f-d2603ecf812d" width="300" />
  <img src="https://github.com/CommunityToolkit/Maui/assets/59332263/39f5b8c9-b8bd-4bf4-b376-b168fc4679f2" width="200" /> 
</p>

## Possible implementation
I'm not sure how to really add it to MAUI since i'm still learning how it works however, I do kinda know how to implement it on Swift. I do know that .NET MAUI is using AVKit to handle playback & such.

On Swift:
```swift
func setNowPlayingMetadata(_ metadata: NowPlayableStaticMetadata) {
       
        let nowPlayingInfoCenter = MPNowPlayingInfoCenter.default()
        var nowPlayingInfo = [String: Any]()

        nowPlayingInfo[MPMediaItemPropertyTitle] = "Song Title"
        nowPlayingInfo[MPMediaItemPropertyArtist] = "The Artist"
        nowPlayingInfo[MPMediaItemPropertyArtwork] = "URL To image"
        nowPlayingInfo[MPMediaItemPropertyAlbumArtist] = "The album artist"
        
        nowPlayingInfoCenter.nowPlayingInfo = nowPlayingInfo //This sets the image to MacOS & iOS
    }
```
On Windows:
```cs
using Windows.Media;

void NowPlaying(string Title, string Artist)
{
        SystemMediaTransportControls systemMediaControls = SystemMediaTransportControls.GetForCurrentView();
        
        systemMediaControls.ButtonPressed += SystemMediaControls_ButtonPressed;
        
        systemMediaControls.DisplayUpdater.Type = MediaPlaybackType.Music;
        systemMediaControls.DisplayUpdater.MusicProperties.Title = "Song Title";
        systemMediaControls.DisplayUpdater.MusicProperties.Artist = "Song Artist";
        systemMediaControls.DisplayUpdater.Update();
}
```

**Comment by ne0rrmatrix** [comment]
Working on hit here: https://github.com/CommunityToolkit/Maui/pull/1782

**Comment by TheCodeTraveler** [comment]
Added in https://github.com/CommunityToolkit/Maui/pull/1782
